From 86bd872294312c906a5e903eab3bdda7c0f41801 Mon Sep 17 00:00:00 2001
From: Alex <alex@dirtyunicorns.com>
Date: Thu, 5 Jul 2018 16:54:29 -0500
Subject: [PATCH] Settings: Change kernel preference behavior

Before you could only click once to see the full kernel version and if
you wanted it to go back to the short version, you had to exit the fragment.

Now you can click between the two (short and long).

Change-Id: I6f89d6b145d5cc7b9cca6febc3651904b98da80c
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 .../KernelVersionPreferenceController.java               | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/com/android/settings/deviceinfo/firmwareversion/KernelVersionPreferenceController.java b/src/com/android/settings/deviceinfo/firmwareversion/KernelVersionPreferenceController.java
index 9116defce16..cc7cbf24721 100644
--- a/src/com/android/settings/deviceinfo/firmwareversion/KernelVersionPreferenceController.java
+++ b/src/com/android/settings/deviceinfo/firmwareversion/KernelVersionPreferenceController.java
@@ -33,6 +33,7 @@ public class KernelVersionPreferenceController extends BasePreferenceController
     private static final String KEY_KERNEL_VERSION = "kernel_version";
     private static final String FILENAME_PROC_VERSION = "/proc/version";
     private static final String LOG_TAG = "KernelVersionPreferenceController";
+    private boolean fullKernelVersion = false;
 
     public KernelVersionPreferenceController(Context context, String preferenceKey) {
         super(context, preferenceKey);
@@ -58,7 +59,13 @@ public boolean handlePreferenceTreeClick(Preference preference) {
         if (!TextUtils.equals(preference.getKey(), KEY_KERNEL_VERSION)) {
             return false;
         }
-        preference.setSummary(getFullKernelVersion());
+        if(fullKernelVersion) {
+            preference.setSummary(DeviceInfoUtils.getFormattedKernelVersion(mContext));
+            fullKernelVersion = false;
+        } else {
+            preference.setSummary(getFullKernelVersion());
+            fullKernelVersion = true;
+        }
         return false;
     }
 
